src/Groups/matrices/transform_form.jl

```diff
@@ -0,0 +1,412 @@
+# TODO: in this file are used many methods TEMPORARILY defined in files matrix_manipulation.jl and stuff_field_gen.jl
+# once methods in those files will be deleted / replaced / modified, this file need to be modified too
+
+
+
+import AbstractAlgebra: Field
+
+export
+    iscongruent
+
+###############################################################################################################
+
+# Algorithm
+
+###############################################################################################################
+
+# The functions find_radical, block_anisotropic_elim and block_herm_elim are based on the article
+# James B. Wilson, Optimal Algorithms of Gram-Schmidt type, Linear Algebra and its Applications 438 (2013) 4573-4583
+
+# returns x,y such that ax^2+by^2 = c
+# at the moment, it simply search for x such that (c-ax^2)/b is a square
+# TODO: is there a faster way to find x and y?
+# TODO: it would be better if this is deterministic. This depends on gen(F) and issquare(F).
+function _solve_eqn(a::T, b::T, c::T) where T <: FieldElem
+   F = parent(a)  
+   ch = Int(characteristic(F))
+   dg = degree(F)
+   w = gen(F)
+   for i in collect(Iterators.product([0:ch-1 for j in 1:dg]...))
+      x = sum([w^(j-1)*i[j] for j in 1:dg])
+      s = (c - a*x^2)*b^-1
+#      vero, y = issquare(s)
+      if issquare(s) return x,square_root(s) end
+   end
+   return nothing, nothing
+end
+
+
+
+# if Symmetric, returns C,A,d where A*B*transpose(frobenius(A,e)) = C, C = block_matrix(2,2,[C,0,0,0]) and d = rank(C)
+# else returns C,A,d where B*A = C, C = [C 0] and d = rank(C)
+# Assumption: if Symmetric==true, then nr=nc always
+function find_radical(B::MatElem{T}, F::Field, nr::Int, nc::Int; e=0, Symmetric=false) where T <: FieldElem
+
+   V1 = VectorSpace(F,nc)
+   V2 = VectorSpace(F,nr)
+   K = Symmetric ? kernel(ModuleHomomorphism(V1,V2,B))[1] : kernel(ModuleHomomorphism(V1,V2,transpose(B)))[1]
+   U,emb = complement(V1,K)
+   d = dim(U)
+   A = matrix(vcat(typeof(emb(gen(U,1)))[emb(v) for v in gens(U)], typeof(K.map(gen(K,1)))[K.map(v) for v in gens(K)] ))
+
+   if Symmetric
+      return A*B*transpose(map(y -> frobenius(y,e),A)), A, d
```

Comment:
Off-topic: we have to multiply by a transpose very often; some systems have dedicate operations for that (which avoid the transpose and directly compute `B*transpose(A)` which saves overhead). Does Julia and/or Oscar offer something like that? I guess it might, since the regular Julia `transpose` is "lazy" by default, so presumably multiplying a regular matrix by one which was lazily transposed can be done efficiently. However, for Nemo matrices, we don't seem to be using that trick...